firmware/esp32_ttgo/client/GPSClient.cpp

```diff
@@ -0,0 +1,19 @@
+#include "GPSClient.h"
+
+void GPSClient::loop()
+{
+    while (stream->available() > 0)
+    {
+        gps.encode(stream->read());
+    }
+
+    int elapsed = millis() - beacon_last;
+    if (elapsed > beacon_period)
+    {
+        if (gps.location.isValid() && gps.location.age() < beacon_period)
+        {
+            server->transmit(this, String("c|<" + username + "> ") + gps.location.lat() + ", " + gps.location.lng());
```

Comment:
Maybe this should be a different message type? `g` or `p` maybe. In case the another client wants to handle the information differently

Reply:
Yeah, I agree, and it should probably be a little more structured. The GPS support was really just a proof of concept.